### PR TITLE
Add stream-monitor binary

### PR DESCRIPTION
**Pull request by thedeeno**
Stream a target file to http ingest in cycles and surface metrics on these stream-cycles via a prom exporter.

### CODE REVIEW

File: cmd/stream-monitor/stream-monitor.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+  hostName, _ := os.Hostname()
+
+  // Print startup
+  fmt.Println("StreamMonitor version: " + model.Version)
+  fmt.Printf("Compiler version: %s %s\n", runtime.Compiler, runtime.Version())
+  fmt.Printf("Hostname %s OS %s IPs %v\n", hostName, runtime.GOOS, utils.GetIPs())
+  fmt.Printf("Production: %v\n", model.Production)
+
+  gctx, _ := context.WithCancel(context.Background()) // to be used as global parent context, in the future
```
Comment (darkdarkdragon) [inline cmd/stream-monitor/stream-monitor.go:69]:
```suggestion
  gctx, gcancel := context.WithCancel(context.Background()) // to be used as global parent context, in the future
  defer gcancel()
```

File: internal/server/metrics.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+type MetricsServer struct {
+}
+
+// NewStreamerServer creates new MetricsServer
+func NewMetricsServer() *MetricsServer {
+	return &MetricsServer{}
+}
+
+// Starts metric server
```
Comment (darkdarkdragon) [inline internal/server/metrics.go:20]:
```suggestion
// Start starts metric server
```

File: internal/server/metrics.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/stream-tester/internal/metrics"
+)
+
+type MetricsServer struct {
+}
+
+// NewStreamerServer creates new MetricsServer
```
Comment (darkdarkdragon) [inline internal/server/metrics.go:15]:
```suggestion
// NewMetricsServer creates new MetricsServer
```

File: internal/server/metrics.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/stream-tester/internal/metrics"
+)
+
+type MetricsServer struct {
```
Comment (darkdarkdragon) [inline internal/server/metrics.go:12]:
```suggestion
// NewMetricsServer object
type MetricsServer struct {
```

**Review by darkdarkdragon — CHANGES_REQUESTED** [review]
Couple of notes.
And please use `gofmt` on all changed files.
I would recommend to use any IDE with Go support - they usually run `gofmt` on file save.

File: cmd/streamtester/streamtester.go [hunk 5]
```diff
@@ -309,7 +309,7 @@ func main() {
 			if len(bds) == 0 {
 				glog.Fatal("Got empty list of broadcasterf from Livepeer API")
 			}
-			up := testers.NewHTTPtreamer(gctx, true, "baseManifestID")
+			up := testers.NewHttpStreamer(gctx, true, "baseManifestID")
```
[the remark this answers is not in the document]
Comment (thedeeno) [reply cmd/streamtester/streamtester.go:312]: Just curious, where did you see this warning? I didn't notice it.

File: internal/server/metrics.go (hunk 2 above)
Comment (thedeeno) [reply internal/server/metrics.go:20]: I think the correction is actually a typo. Do you mind if I keep the original?

File: internal/server/metrics.go (hunk 4 above)
Comment (thedeeno) [reply internal/server/metrics.go:12]: hmm, I find this comment redundant. Is this a convention we're following?

**Comment by thedeeno** [comment]
Re: gofmt, will do. I'll be sure I have this formatter enabled in my vim.

**Comment by thedeeno** [comment]
Thanks for the review!

File: cmd/streamtester/streamtester.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (darkdarkdragon) [reply cmd/streamtester/streamtester.go:312]: run `golint`

File: internal/server/metrics.go (hunk 4 above)
Comment (darkdarkdragon) [reply internal/server/metrics.go:12]: `golint internal/server/metrics.go`

File: internal/server/metrics.go (hunk 2 above)
Comment (darkdarkdragon) [reply internal/server/metrics.go:20]: `golint internal/server/metrics.go`

**Review by darkdarkdragon — CHANGES_REQUESTED** [review]

**Comment by thedeeno** [comment]
Ok great updated. I didn't realize I disabled my linters/formatters.

**Comment by thedeeno** [comment]
it might make sense to add those checks as a github action on PR yeah?

**Comment by darkdarkdragon** [comment]
> it might make sense to add those checks as a github action on PR yeah?

No, not now

**Review by darkdarkdragon — APPROVED** [review]
LGTM

**Pull request merged**